cpp/src/IceStorm/TopicI.cpp

```diff
@@ -131,14 +135,15 @@ class TopicI : public TopicInternal
             {
                 try
                 {
-                    return master->subscribeAndGetPublisher(std::move(qos), std::move(obj));
+                    // Don't move the parameters as we may need to retry.
+                    return master->subscribeAndGetPublisher(qos, *obj);
```

Comment:
The call to std::move here was bogus, as the parameter can be used again if we retry, see exception handler below.

Reply:
good catch